src/main/java/com/minecolonies/coremod/inventory/InventoryCitizen.java

```diff
@@ -586,136 +302,31 @@ public void markDirty()
      * @param player the player acessing the inventory.
      * @return if the player is allowed to access.
      */
-    @Override
     public boolean isUseableByPlayer(@NotNull final EntityPlayer player)
     {
         return this.citizen.getColony().getPermissions().hasPermission(player, Permissions.Action.ACCESS_HUTS);
     }
 
     /**
-     * Called when inventory is opened by a player.
-     *
-     * @param player the player who opened the inventory.
-     */
-    @Override
-    public void openInventory(final EntityPlayer player)
-    {
-        /*
-         * This may be filled in order to specify some custom handling.
-         */
-    }
-
-    /**
-     * Called after the inventory has been closed by a player.
-     *
-     * @param player the player who opened the inventory.
-     */
-    @Override
-    public void closeInventory(final EntityPlayer player)
-    {
-        /*
-         * This may be filled in order to specify some custom handling.
-         */
-    }
-
-    /**
-     * Returns true if automation is allowed to insert the given stack (ignoring stack size) into the given slot.
-     *
-     * @param index the accessing slot.
-     * @param stack the stack trying to enter.
-     * @return if the stack may be inserted.
-     */
-    @Override
-    public boolean isItemValidForSlot(final int index, final ItemStack stack)
-    {
-        return true;
-    }
-
-    /**
-     * This may be used in order to return values of different GUI areas like the ones in the beacon.
-     *
-     * @param id the id of the field.
-     * @return the value of the field.
-     */
-    @Override
-    public int getField(final int id)
-    {
-        return 0;
-    }    /**
      * Get the formatted TextComponent that will be used for the sender's username in chat.
      */
     @NotNull
-    @Override
     public ITextComponent getDisplayName()
     {
         return this.hasCustomName() ? new TextComponentString(this.getName()) : new TextComponentTranslation(this.getName());
     }
 
     /**
-     * This may be used to set GUI areas with a certain id and value.
-     *
-     * @param id    some id.
-     * @param value some value.
-     */
-    @Override
-    public void setField(final int id, final int value)
-    {
-        /*
-         * We currently need no fields.
-         */
-    }
-
-    /**
-     * Returns the number of fields.
-     *
-     * @return the amount.
-     */
-    @Override
-    public int getFieldCount()
-    {
-        return 0;
-    }
-
-    /**
-     * Completely clears the inventory.
-     */
-    @Override
-    public void clear()
-    {
-        for (int i = 0; i < this.stacks.length; ++i)
-        {
-            this.stacks[i] = null;
-        }
-    }
-
-    /**
      * Used to store variables.
-     *
-     * @param compound with the given tag.
      */
-    public void writeToNBT(@NotNull final NBTTagCompound compound)
+    public NBTTagCompound serializeNBT()
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Add the "@Override" annotation above this method signature [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1161)